src/core/weights.js

```diff
@@ -165,3 +166,90 @@ export function toJSON(weights: Weights): WeightsJSON {
 export function fromJSON(json: WeightsJSON): Weights {
   return parser.parseOrThrow(json);
 }
+
+export type NodeWeightDiff = {|
+  +address: NodeAddressT,
+  +first: NodeWeight | null,
+  +second: NodeWeight | null,
+|};
+
+export type EdgeWeightDiff = {|
+  +address: EdgeAddressT,
+  +first: EdgeWeight | null,
+  +second: EdgeWeight | null,
+|};
+
+export type WeightsComparison = {|
+  +weightsAreEqual: boolean,
+  +nodeWeightDiffs: $ReadOnlyArray<NodeWeightDiff>,
+  +edgeWeightDiffs: $ReadOnlyArray<EdgeWeightDiff>,
+|};
+
+export function compareWeights(
+  firstWeights: Weights,
+  secondWeights: Weights
+): WeightsComparison {
+  const nodeWeightDiffs = [];
+  const edgeWeightDiffs = [];
+  const weightsAreEqual = deepEqual(firstWeights, secondWeights);
```

Comment:
I'm not sure this is needed, feels like an unecessary performance optimization. Also, off the top of my head I'm not 100% sure that `new Map().set("a", 1).set("b", 2)` is deep equal to `new Map().set("b", 2).set("a", 1)`. Since afaik maps do iterate in insertion order, there is an observable diff between the two. But probably deep equal would return true?

Reply:
yeah from my tests, the deep equals is order-agnostic. it does seem like an opportunity for unexpected behavior to creep in, though, and I only added it on your request, so I'm happy to take it out.

Reply:
Yeah, in the case of graph comparison, there was more overhead in iterating over all the nodes v.s. using the `graph.equals` method which is a lot more efficient. Here the overhead is a lot smaller. But I think in either case, my suggestion was a premature optimization that added complexity in a case that's very unlikely to be the perf-critical path.